[BUG] cudf JAVA build failed "cannot find cudfConfig.cmake and cudf-config.cmake"

**Describe the bug**
```
[2025-05-16T03:19:06.102Z]      [exec] -- Using pre-configured libcudf from build path: 
[2025-05-16T03:19:06.102Z]      [exec] -- Configuring incomplete, errors occurred!
[2025-05-16T03:19:06.102Z]      [exec] CMake Error at /home/jenkins/agent/workspace/jenkins-cudf_nightly-dev-github-1529-cuda11/java/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:125 (find_package):
[2025-05-16T03:19:06.102Z]      [exec]   By not providing "Findcudf.cmake" in CMAKE_MODULE_PATH this project has
[2025-05-16T03:19:06.102Z]      [exec]   asked CMake to find a package configuration file provided by "cudf", but
[2025-05-16T03:19:06.102Z]      [exec]   CMake did not find one.
[2025-05-16T03:19:06.102Z]      [exec] 
[2025-05-16T03:19:06.102Z]      [exec]   Could not find a package configuration file provided by "cudf" with any of
[2025-05-16T03:19:06.102Z]      [exec]   the following names:
[2025-05-16T03:19:06.102Z]      [exec] 
[2025-05-16T03:19:06.102Z]      [exec]     cudfConfig.cmake
[2025-05-16T03:19:06.102Z]      [exec]     cudf-config.cmake
[2025-05-16T03:19:06.102Z]      [exec] 
[2025-05-16T03:19:06.102Z]      [exec]   Add the installation prefix of "cudf" to CMAKE_PREFIX_PATH or set
[2025-05-16T03:19:06.102Z]      [exec]   "cudf_DIR" to a directory containing one of the above files.  If "cudf"
[2025-05-16T03:19:06.102Z]      [exec]   provides a separate development package or SDK, be sure it has been
[2025-05-16T03:19:06.102Z]      [exec]   installed.
[2025-05-16T03:19:06.102Z]      [exec] Call Stack (most recent call first):
[2025-05-16T03:19:06.102Z]      [exec]   CMakeLists.txt:95 (rapids_find_package)
[2025-05-16T03:19:06.102Z]      [exec] 
[2025-05-16T03:19:06.102Z]      [exec] 
[2025-05-16T03:19:06.102Z] [INFO] ------------------------------------------------------------------------
[2025-05-16T03:19:06.102Z] [INFO] BUILD FAILURE
[2025-05-16T03:19:06.102Z] [INFO] ------------------------------------------------------------------------
```

**Steps/Code to reproduce bug**
https://github.com/rapidsai/cudf/blob/branch-25.06/java/ci/build-in-docker.sh

**Expected behavior**
Pass the build and test

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source
   - https://github.com/rapidsai/cudf/blob/branch-25.06/java/ci/Dockerfile.rocky

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**Additional context**
Add any other context about the problem here.

--- a/java/ci/build-in-docker.sh
+++ b/java/ci/build-in-docker.sh
@@ -95,7 +95,7 @@ if [ -f "$WORKSPACE/java/ci/settings.xml" ]; then
 fi
 
 cd "$WORKSPACE/java"
-CUDF_INSTALL_DIR="$LIBCUDF_BUILD_PATH" mvn -B clean package $BUILD_ARG
+CUDF_INSTALL_DIR="$INSTALL_PREFIX" mvn -B clean package $BUILD_ARG
 
 ###### Stash Jar files ######
 rm -rf $OUT_PATH

--- a/java/ci/build-in-docker.sh
+++ b/java/ci/build-in-docker.sh
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 #
-# Copyright (c) 2020-2024, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020-2025, NVIDIA CORPORATION. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -51,9 +51,10 @@ export CUDACXX=/usr/local/cuda/bin/nvcc
 export LIBCUDF_KERNEL_CACHE_PATH=/rapids
 
 ###### Build libcudf ######
-rm -rf "$WORKSPACE/cpp/build"
-mkdir -p "$WORKSPACE/cpp/build"
-cd "$WORKSPACE/cpp/build"
+LIBCUDF_BUILD_PATH="$WORKSPACE/cpp/build"
+rm -rf "$LIBCUDF_BUILD_PATH"
+mkdir -p "$LIBCUDF_BUILD_PATH"
+cd "$LIBCUDF_BUILD_PATH"
 cmake .. -G"${CMAKE_GENERATOR}" \
          -DCMAKE_INSTALL_PREFIX=$INSTALL_PREFIX \
          -DCUDA_STATIC_RUNTIME=$ENABLE_CUDA_STATIC_RUNTIME \
@@ -65,7 +66,8 @@ cmake .. -G"${CMAKE_GENERATOR}" \
          -DCUDF_USE_PER_THREAD_DEFAULT_STREAM=$ENABLE_PTDS \
          -DRMM_LOGGING_LEVEL=$RMM_LOGGING_LEVEL \
          -DBUILD_SHARED_LIBS=OFF \
-         -DCUDF_KVIKIO_REMOTE_IO=OFF
+         -DCUDF_KVIKIO_REMOTE_IO=OFF \
+         -DCUDF_EXPORT_NVCOMP=ON
 
 if [[ -z "${PARALLEL_LEVEL}" ]]; then
     cmake --build .
@@ -93,7 +95,7 @@ if [ -f "$WORKSPACE/java/ci/settings.xml" ]; then
 fi
 
 cd "$WORKSPACE/java"
-mvn -B clean package $BUILD_ARG
+CUDF_INSTALL_DIR="$LIBCUDF_BUILD_PATH" mvn -B clean package $BUILD_ARG
 
 ###### Stash Jar files ######
 rm -rf $OUT_PATH

